src/chainstate/burn/db/sortdb.rs

```diff
@@ -3695,6 +3705,7 @@ impl<'a> SortitionHandleTx<'a> {
             &snapshot.accumulated_coinbase_ustx.to_string(),
         ];
 
+        // Note: insert into snap shots.. is this done by the time that we have this block?
```

Comment:
"snapshots"

Yes, this happens once we have processed and stored the bitcoin block and extracted and processed all of its stacks-specific transactions.  There is one snapshot per sortition, which is a view calculated over the set of block-commits in a bitcoin block.